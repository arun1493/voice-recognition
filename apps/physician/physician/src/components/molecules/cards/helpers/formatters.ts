import i18n from "i18next";
import _get from "lodash/get";
import { EMPTY_OBJECT } from "@suki-web/suki-business";

import {
  Allergy,
  AllergyReaction,
  CardMedicalHistoryProps,
  CardSurgicalHistoryProps,
  ParagraphCard,
  Problem
} from "../../../../types";
import { CardType } from "../../../../constants";

const formatParagraphCardData = (data: CardMedicalHistoryProps | CardSurgicalHistoryProps | any, cardType: string) => {
  const paragraphData: Array<ParagraphCard> = [];

  switch (cardType) {
    case CardType.ALLERGIES: {
      const { allergies, nkda } = data || EMPTY_OBJECT;

      allergies &&
        allergies.forEach((allergyItem: Allergy) => {
          const { allergen_name, reactions } = allergyItem || EMPTY_OBJECT;
          let allergyReactionContent = "";
          reactions &&
            reactions.forEach((allergyItem: AllergyReaction) => {
              const { reaction_name } = allergyItem;
              if (reaction_name) allergyReactionContent = allergyReactionContent.concat(reaction_name, ", ");
            });

          allergyReactionContent = allergyReactionContent.slice(0, -2); // to remove comma at the end of string

          if (allergen_name)
            paragraphData.push({
              title: allergen_name,
              content: allergyReactionContent
            });
        });

      if (nkda)
        paragraphData.push({
          content: i18n.t("pages.note.cards.errors.allergiesNkda")
        });
      break;
    }
    case CardType.PAST_MEDICAL_HISTORY: {
      const { active_problems } = data || EMPTY_OBJECT;
      active_problems &&
        active_problems.forEach((activeProblem: Problem) => {
          const description = _get(activeProblem, "note");
          const error = _get(activeProblem, "error");
          const title = _get(activeProblem, "diagnosis.icd_description");
          if (!error) paragraphData.push({ title, content: description });
        });
      break;
    }

    case CardType.PAST_SURGICAL_HISTORY: {
      data &&
        data.forEach((surgicalHistoryItem: Problem) => {
          const { description, note, error } = surgicalHistoryItem;
          if (!error) paragraphData.push({ title: description, content: note });
        });
      break;
    }

    default:
      break;
  }
  return paragraphData;
};

export { formatParagraphCardData };
