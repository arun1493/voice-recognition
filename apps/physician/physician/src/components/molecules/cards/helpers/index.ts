export * from "./labels";
export * from "./formatters";
