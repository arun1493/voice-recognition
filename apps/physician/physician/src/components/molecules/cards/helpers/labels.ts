import i18n from "i18next";

import { CardType, VITAL_SIGNS } from "../../../../constants";

const getCardTitle = (cardType: string) => {
  switch (cardType) {
    case CardType.ALLERGIES:
      return i18n.t("pages.note.cards.titles.allergies");
    case CardType.CURRENT_MEDICATION:
      return i18n.t("pages.note.cards.titles.medication");
    case CardType.PAST_MEDICAL_HISTORY:
    case CardType.PAST_MEDICAL_HISTORY_SHOW_ME:
      return i18n.t("pages.note.cards.titles.medicalHistory");
    case CardType.PAST_SURGICAL_HISTORY:
    case CardType.PAST_SURGICAL_HISTORY_SHOW_ME:
      return i18n.t("pages.note.cards.titles.surgicalHistory");
    case CardType.VITALS:
      return i18n.t("pages.note.cards.titles.vitals");
    default:
      return "";
  }
};

const getVitalSign = (sign: string) => {
  switch (sign) {
    case VITAL_SIGNS.BLOOD_PRESSURE:
      return i18n.t("pages.note.cards.vitalSigns.bloodPressure");
    case VITAL_SIGNS.BMI:
      return i18n.t("pages.note.cards.vitalSigns.bmi");
    case VITAL_SIGNS.HEIGHT:
      return i18n.t("pages.note.cards.vitalSigns.height");
    case VITAL_SIGNS.PULSE:
      return i18n.t("pages.note.cards.vitalSigns.pulse");
    case VITAL_SIGNS.WEIGHT:
      return i18n.t("pages.note.cards.vitalSigns.weight");
    case VITAL_SIGNS.HEART_RATE:
      return i18n.t("pages.note.cards.vitalSigns.heartRate");
    case VITAL_SIGNS.RESPIRATORY_RATE:
      return i18n.t("pages.note.cards.vitalSigns.respiratoryRate");
    case VITAL_SIGNS.TEMPERATURE:
      return i18n.t("pages.note.cards.vitalSigns.temperature");
    default:
      return null;
  }
};

const getErrorMessage = (cardType: string) => {
  switch (cardType) {
    case CardType.ALLERGIES:
      return i18n.t("pages.note.cards.errors.allergies");
    case CardType.CURRENT_MEDICATION:
      return i18n.t("pages.note.cards.errors.medication");
    case CardType.PAST_MEDICAL_HISTORY:
    case CardType.PAST_MEDICAL_HISTORY_SHOW_ME:
      return i18n.t("pages.note.cards.errors.medicalHistory");
    case CardType.PAST_SURGICAL_HISTORY:
    case CardType.PAST_SURGICAL_HISTORY_SHOW_ME:
      return i18n.t("pages.note.cards.errors.surgicalHistory");
    case CardType.VITALS:
      return i18n.t("pages.note.cards.errors.vitals");
    default:
      return "";
  }
};

export { getCardTitle, getErrorMessage, getVitalSign };
