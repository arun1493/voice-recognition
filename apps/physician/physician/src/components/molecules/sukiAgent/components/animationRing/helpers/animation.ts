import { RING_ANIMATION_DURATION, RING_BASE_OPACITY, RING_SCALE, SukiAgentState } from "../../../../../../constants";

function isValidSukiAgentStateForSingleRingAnimation(agentStatus: SukiAgentState, isAgentActive: boolean) {
  return (isAgentActive && agentStatus === SukiAgentState.AWAKE) || agentStatus === SukiAgentState.PROCESSING;
}

function isValidSukiAgentStateForMultipleRingAnimation(agentStatus: SukiAgentState, isAgentActive: boolean) {
  return (
    isAgentActive &&
    !isValidSukiAgentStateForSingleRingAnimation(agentStatus, isAgentActive) &&
    agentStatus === SukiAgentState.LISTENING
  );
}

function getDictationRingVariants(ringsCount: number) {
  const dictationRingVariants = [];
  for (let i = ringsCount; i > 0; i--) {
    const RING_VARIANTS = {
      oscillate: {
        opacity: RING_BASE_OPACITY - i * 0.2,
        scale: RING_SCALE + i * 0.01,
        transition: {
          yoyo: Infinity,
          duration: RING_ANIMATION_DURATION,
          type: "tween",
          delay: i * 0.1,
          ease: "easeInOut"
        }
      }
    };

    dictationRingVariants.push(RING_VARIANTS);
  }
  return dictationRingVariants;
}

export {
  getDictationRingVariants,
  isValidSukiAgentStateForMultipleRingAnimation,
  isValidSukiAgentStateForSingleRingAnimation
};
