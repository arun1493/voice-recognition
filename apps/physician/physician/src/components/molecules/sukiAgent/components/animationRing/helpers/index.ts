export * from "./ringSize";
export * from "./animation";
