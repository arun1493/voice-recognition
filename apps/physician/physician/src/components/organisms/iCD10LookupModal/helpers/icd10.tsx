import _isEmpty from "lodash/isEmpty";

const formatText = (text: string, highlight: string) => {
  if (_isEmpty(text)) return;
  const parts = text.split(new RegExp(`(${highlight})`, "gi"));
  return parts.map((part) => (part.toLowerCase() === highlight.toLowerCase() ? <strong>{part}</strong> : part));
};

const formatCode = (code: string) => {
  if (code?.length > 3) {
    return code?.substring(0, 3) + "." + code?.substring(3);
  }
  return code;
};

export { formatCode, formatText };
