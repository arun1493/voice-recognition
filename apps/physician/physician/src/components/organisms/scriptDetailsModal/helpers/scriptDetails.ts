import { ScreenSize } from "@suki-web/suki-ui";

const { SM, XS, XXS } = ScreenSize;

const getWidth = (breakPoint: ScreenSize) => {
  switch (breakPoint) {
    case XXS:
      return "100vw";
    case XS:
      return 484;
    case SM:
      return 664;
    default:
      return 782;
  }
};

const getHeight = (breakPoint: ScreenSize) =>
  breakPoint === XS || breakPoint === XXS ? "100vh" : "calc( 100vh - 56px )";

const getMaxHeight = (breakPoint: ScreenSize) => (breakPoint === XS || breakPoint === XXS ? "100vh" : 0);

export { getHeight, getMaxHeight, getWidth };
