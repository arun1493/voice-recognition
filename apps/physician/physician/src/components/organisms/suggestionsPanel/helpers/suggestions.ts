import i18n from "i18next";

import { SuggestionTypes } from "../../../../constants";

const getTextBySuggestion = (suggestion: string) => {
  switch (suggestion) {
    case SuggestionTypes.PL:
      return i18n.t("components.suggestionsPanel.pl");
    case SuggestionTypes.ED:
      return i18n.t("components.suggestionsPanel.ed");
    default:
      return "";
  }
};

export { getTextBySuggestion };
