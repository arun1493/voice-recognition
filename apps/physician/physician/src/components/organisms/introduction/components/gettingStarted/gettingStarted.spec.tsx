import "@testing-library/jest-dom/extend-expect";
import { cleanup, RenderResult } from "@testing-library/react";

import { GettingStarted } from ".";
import { configureI18n } from "../../../../../helpers";
import { componentSetupWithStore } from "../../../../../hocs";
import { shouldMatchSnapshot } from "../../../../../helpers/snapshot.spec";

const setup = componentSetupWithStore(() => <GettingStarted />);

describe("Getting Started", () => {
  let component: RenderResult;

  beforeEach(() => {
    configureI18n();
    component = setup();
  });

  afterEach(() => {
    cleanup();
  });

  it("should render successfully", () => {
    const { baseElement } = component;
    expect(baseElement).toBeTruthy();
  });

  it("should render the correct heading", () => {
    const { baseElement } = component;
    expect(baseElement.getElementsByTagName("h1")[0].innerHTML).toEqual("Getting Started");
  });

  it("should render the correct desription", () => {
    const { baseElement } = component;
    expect(baseElement.querySelector("[role=doc-subtitle]")?.innerHTML).toEqual("Try these actions or voice commands");
  });

  it("should match snapshot", () => {
    shouldMatchSnapshot(GettingStarted);
  });
});
