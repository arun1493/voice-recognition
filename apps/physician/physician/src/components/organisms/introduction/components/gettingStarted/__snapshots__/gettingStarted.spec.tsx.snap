// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Getting Started should match snapshot 1`] = `
<div
  className="gettingStartedstyled__StyledGettingStarted-sc-1k5igtd-1 kOdaww"
>
  <div
    className="gettingStartedstyled__HeaderContainer-sc-1k5igtd-0 fJNunX"
  >
    <h1
      className="headerstyled__StyledH1-sc-1y4xvz5-0 ifdpFf title"
      data-cy="intro-screen4-header"
    >
      Getting Started
    </h1>
    <div
      data-cy="intro-screen4-description"
      role="doc-subtitle"
    >
      Try these actions or voice commands
    </div>
  </div>
  <div
    className="gettingStartedstyled__StyledTaskList-sc-1k5igtd-2 VkVLL"
    data-cy="intro-screen4-list"
  >
    <div
      className="taskstyled__StyledTask-sc-1n5gkgl-0 krPpMb"
    >
      <div
        className="taskstyled__StyledItem-sc-1n5gkgl-1 fjArXj"
      >
        <svg
          height={24}
          viewBox="0 0 24 24"
          width={24}
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="11"
            cy="11"
            fill="#E6E6E6"
            fillRule="evenodd"
            opacity="1"
            r="11"
            transform="translate(1 1)"
          />
        </svg>
        <p
          className="textstyled__StyledText-sc-1vedkxi-0 fDyqSY"
          color="inherit"
          cursor="default"
          onClick={[Function]}
          size="md"
        >
          Turn Suki badge off and on
        </p>
      </div>
      <div
        className="taskstyled__StyledChildrenContainer-sc-1n5gkgl-2 ehrpBb"
      />
    </div>
    <div
      className="taskstyled__StyledTask-sc-1n5gkgl-0 krPpMb"
    >
      <div
        className="taskstyled__StyledItem-sc-1n5gkgl-1 fjArXj"
      >
        <svg
          height={24}
          viewBox="0 0 24 24"
          width={24}
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="11"
            cy="11"
            fill="#E6E6E6"
            fillRule="evenodd"
            opacity="1"
            r="11"
            transform="translate(1 1)"
          />
        </svg>
        <p
          className="textstyled__StyledText-sc-1vedkxi-0 fDyqSY"
          color="inherit"
          cursor="default"
          onClick={[Function]}
          size="md"
        >
          Play Demo
        </p>
      </div>
      <div
        className="taskstyled__StyledChildrenContainer-sc-1n5gkgl-2 ehrpBb"
      />
    </div>
    <div
      className="taskstyled__StyledTask-sc-1n5gkgl-0 krPpMb"
    >
      <div
        className="taskstyled__StyledItem-sc-1n5gkgl-1 fjArXj"
      >
        <svg
          height={24}
          viewBox="0 0 24 24"
          width={24}
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="11"
            cy="11"
            fill="#E6E6E6"
            fillRule="evenodd"
            opacity="1"
            r="11"
            transform="translate(1 1)"
          />
        </svg>
        <p
          className="textstyled__StyledText-sc-1vedkxi-0 fDyqSY"
          color="inherit"
          cursor="default"
          onClick={[Function]}
          size="md"
        >
          "Suki, What can I say ?"
        </p>
      </div>
      <div
        className="taskstyled__StyledChildrenContainer-sc-1n5gkgl-2 ehrpBb"
      />
    </div>
    <div
      className="taskstyled__StyledTask-sc-1n5gkgl-0 krPpMb"
    >
      <div
        className="taskstyled__StyledItem-sc-1n5gkgl-1 fjArXj"
      >
        <svg
          height={24}
          viewBox="0 0 24 24"
          width={24}
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="11"
            cy="11"
            fill="#E6E6E6"
            fillRule="evenodd"
            opacity="1"
            r="11"
            transform="translate(1 1)"
          />
        </svg>
        <p
          className="textstyled__StyledText-sc-1vedkxi-0 fDyqSY"
          color="inherit"
          cursor="default"
          onClick={[Function]}
          size="md"
        >
          Tap Patients to see your schedule or patient list
        </p>
      </div>
      <div
        className="taskstyled__StyledChildrenContainer-sc-1n5gkgl-2 ehrpBb"
      />
    </div>
    <div
      className="taskstyled__StyledTask-sc-1n5gkgl-0 krPpMb"
    >
      <div
        className="taskstyled__StyledItem-sc-1n5gkgl-1 fjArXj"
      >
        <svg
          height={24}
          viewBox="0 0 24 24"
          width={24}
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="11"
            cy="11"
            fill="#E6E6E6"
            fillRule="evenodd"
            opacity="1"
            r="11"
            transform="translate(1 1)"
          />
        </svg>
        <p
          className="textstyled__StyledText-sc-1vedkxi-0 fDyqSY"
          color="inherit"
          cursor="default"
          onClick={[Function]}
          size="md"
        >
          Complete 5 notes in Suki
        </p>
      </div>
      <div
        className="taskstyled__StyledChildrenContainer-sc-1n5gkgl-2 ehrpBb"
      >
        <div
          className="progressBarstyled__StyledBarContainer-sc-1ryqtdv-0 cwPInP"
          data-cy="intro-screen4-progress-bar"
        >
          <div
            className="progressBarstyled__StyledProgressItem-sc-1ryqtdv-1 progressBarstyled__StyledPendingItem-sc-1ryqtdv-2 fwFsfl bcfOoZ"
          />
          <div
            className="progressBarstyled__StyledProgressItem-sc-1ryqtdv-1 progressBarstyled__StyledPendingItem-sc-1ryqtdv-2 fwFsfl bcfOoZ"
          />
          <div
            className="progressBarstyled__StyledProgressItem-sc-1ryqtdv-1 progressBarstyled__StyledPendingItem-sc-1ryqtdv-2 fwFsfl bcfOoZ"
          />
          <div
            className="progressBarstyled__StyledProgressItem-sc-1ryqtdv-1 progressBarstyled__StyledPendingItem-sc-1ryqtdv-2 fwFsfl bcfOoZ"
          />
          <div
            className="progressBarstyled__StyledProgressItem-sc-1ryqtdv-1 progressBarstyled__StyledPendingItem-sc-1ryqtdv-2 fwFsfl bcfOoZ"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
